jni/include/knn_extension/faiss/utils/BitSet.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <faiss/MetricType.h>
+#include <faiss/impl/platform_macros.h>
+#include <limits>
+
+using idx_t = faiss::idx_t;
+/**
+ * This class is used to store parent and child doc id mapping
+ *
+ * For example, let's say there are two documents with 3 nested field each. Then, lucene store each nested field as
+ * individual document with its own doc id. The document ids are assigned as following.
+ *
+ * 0, 1, 2, 3(parent doc for 0, 1, 2), 4, 5, 6, 7(parent doc for 4, 5, 6)
+ *
+ * Therefore, we can represent the value in BitSet like 10001000 where parent doc id position is set as 1
+ * and child doc id position is set as 0. Finally, by using nextSetBit method, we can find parent ID of a
+ * given document ID.
+ */
+class BitSet {
+protected:
+    const int NO_MORE_DOCS = std::numeric_limits<int>::max();
+public:
+    virtual idx_t nextSetBit(idx_t index) = 0;
+    virtual ~BitSet() = default;
+};
+
+
+/**
+ * BitSet implementation by using an array of unit64
+ */
+class FixedBitSet : public BitSet {
+public:
+    size_t n;
```

Comment:
I assume n is the max number of docs in the set?

Reply:
Let me add a comment. It is a length of bitmap.